File: BoyGirl.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    string male = "IGNORE HIM!\n";
    string female = "CHAT WITH HER!\n";
    // taking input
    string userName;
    cin >> userName;
    // using set, since we only need unique characters from the username
    set<char> s1;
    for (auto i=userName.begin(); i!=userName.end(); i++) {
        s1.insert(*i);
    }
    int result = s1.size();
    // self-explanatory
    if (result%2==0)
        cout << female;
    else 
        cout << male;

    return 0;
}

File: Python/BoyGirl.py
def main():
    male = "IGNORE HIM!\n"
    female = "CHAT WITH HER\n"
    username = input()
    s1 = {i for i in username}
    print(female if len(s1) % 2 == 0 else male)


if __name__ == "__main__":
    main()
